src/pages/spot/Spot.tsx

```diff
@@ -1,36 +1,249 @@
+    if (!tokenMetadata || tokenPrice == null) return null;
+
+    return {
+      name: tokenMetadata.tokenNameFull ?? tokenMetadata.symbol ?? 'Unknown',
+      symbol: tokenMetadata.symbol ?? 'Unknown',
+      tokenAddress: tokenMint!,
+      buys24hUsd: tokenMetadata.token24hBuys,
+      sells24hUsd: -MustNumber(tokenMetadata.token24hSells),
+      change24hPercent: tokenMetadata.pricePercentChange24h,
+      circulatingSupply: MustNumber(tokenMetadata.circulatingSupply),
+      liquidityUsd: tokenMetadata.liquidityUSD,
+      logoUrl: tokenMetadata.image,
+      marketCapUsd: MustNumber(tokenMetadata.circulatingSupply) * tokenPrice,
+      fdvUsd: MustNumber(tokenMetadata.totalSupply) * tokenPrice,
+      priceUsd: tokenPrice,
+      totalSupply: MustNumber(tokenMetadata.totalSupply),
+      volume24hUsd: tokenMetadata.volumeUSD,
+      holders: tokenMetadata.holders,
+      top10HoldersPercent: mapIfPresent(tokenMetadata.top10HoldersPercent, (v) => v / 100),
+      devHoldingPercent: mapIfPresent(tokenMetadata.devHoldingPercent, (v) => v / 100),
+      snipersPercent: mapIfPresent(tokenMetadata.snipersPercent, (v) => v / 100),
+      bundlersPercent: mapIfPresent(tokenMetadata.bundlersPercent, (v) => v / 100),
+      insidersPercent: mapIfPresent(tokenMetadata.insidersPercent, (v) => v / 100),
+    };
+  }, [tokenMint, tokenMetadata, tokenPrice]);
+
+  const handleTokenSelect = (token: SpotHeaderToken) => {
+    navigate(`/spot/${token.tokenAddress}`);
+    setSearchQuery('');
+  };
+
+  const handlePositionSelect = (token: SpotPositionItem) => {
+    navigate(`/spot/${token.tokenAddress}`);
+  };
+
+  const handleTokenSearchChange = (value: string) => {
+    setSearchQuery(value);
+  };
+
+  const handlePositionSell = (token: SpotPositionItem) => {
+    dispatch(spotFormActions.setSide(SpotSide.SELL));
+    navigate(`/spot/${token.tokenAddress}`);
+  };
+
   return (
-    <$SpotLayout tradeLayout={tradeLayout}>
+    <$SpotLayout tradeLayout={tradeLayout} isHorizontalOpen={isHorizontalOpen}>
       <header tw="[grid-area:Top]">
-        <div tw="p-1">Spot Market Selector (Coming Soon)</div>
+        <SpotHeader
+          currentToken={currentTokenData}
+          searchResults={searchResults}
+          isSearchLoading={isSearchLoading}
+          isTokenLoading={isTokenMetadataLoading}
+          onTokenSelect={handleTokenSelect}
+          onSearchTextChange={handleTokenSearchChange}
+        />
       </header>
 
       <$GridSection gridArea="Side">
-        <div tw="p-1">Spot Side Panel (Coming Soon)</div>
+        <SpotTradeForm />
+        <SpotTokenInfo
+          isLoading={isTokenMetadataLoading}
+          links={[
+            { icon: IconName.Earth, url: '1' },
+            { icon: IconName.File, url: '2' },
+            { icon: IconName.CoinMarketCap, url: '3' },
+            { icon: IconName.SocialX, url: '4' },
+          ]}
+          contractAddress={tokenMint!}
```

Comment:
nit: Maybe we should have an early return with messaging so we don't need to `tokenMint!` here or below. I understand Spot page only appears if param is present